### DISCUSSION TITLE
Control of the Magene L508

### DISCUSSION THREAD

**Discussion by izumina0** [maca88/SmartBikeLights#57] [Q&A]
I have a question about the connection between the Magene L508 and the Edge840Solor.
The L508's default settings in the Lights Configurator (LC) did not work correctly, the version of the L508 is 0.114.
Controls from the Edge side were not accepted, and when the light mode was changed using a switch on the L508 itself, the display on the Edge side changed, but the actual lights and names were different.
[LC name=L508 light]
Solid=Solid
Flash=Palse
Palse=(Unknown)
Peloton=Peloton
Rotate=(Unknown)
Quick Flash=Flash

Is there a solution?

**Comment by maca88** [comment]
Hi,

Can you try to clear the "Lights Configuration" app setting and report what numbers do you see on each button along with the light mode that is activated when the number is pressed? For example something like this:

1 -> Solid
2 -> Low
8 -> Pulse
...

Perhaps Magene changed the light modes number in the recent firmware. Unfortunately I don't have a Magene light to test it out.

**Reply by izumina0** [reply]
Checked the numbers.
Solid=4
Flashing=6
Pulse=?
Peloton=5
Rotation=?
Quickly flash=7

Some modes do not display numbers. Is there a way to find out?

**Reply by maca88** [reply]
In order to find the light modes for the missing modes you will need to use the LightTester app that I created for such cases. If you are willing to install the LightTester app, you can follow the instructions described here:
https://github.com/maca88/SmartBikeLights/tree/tester/Source/LightTester

I uploaded the `.prg` file for Edge 840 here: https://github.com/maca88/SmartBikeLights/blob/tester/Source/LightTester/build/edge840/LightTester.prg

It is very important that in step 8 you go though all modes by pressing the physical button on the light and write down also the order of the light modes when switching them by using the light button. So that when I will check the logs of the light modes I will be able to identify what number represents what mode.

**Reply by izumina0** [reply]
Hi.
I have logged.

Result.
Solid = 4
Flashing=6
Pulse=48
Peloton=5
Rotation=49
Quickly flash=7

Attach txt file.
[LightTester.txt](https://github.com/maca88/SmartBikeLights/files/14035659/LightTester.txt)

**Reply by maca88** [reply]
Thanks for posting the results. Based on the results, I can see two issues in the L508 firmware:

1. Pulse and Rotation light modes should be numbered as 63 & 62 instead of 48 & 49
2. Only one custom mode is exposed (63), which would be Pulse if the numbers would be correct (first issue)

I wrote to Magene support pointing out these issues, so let's hope that they will acknowledge and fix the issues in a future firmware release. If not, we will need to find an alternative solution. For now, let's wait for a response from Magene. I will update you when I have more information.

**Reply by izumina0** [reply]
Understood.
I wish you good results. It taught me a lot and gave me a good experience.

Much obliged, maca88.
